packages/pwa-kit-react-sdk/src/ssr/browser/main.jsx

```diff
@@ -118,7 +118,7 @@ export const start = () => {
     return Promise.resolve()
         .then(() => new Promise((resolve) => loadableReady(resolve)))
         .then(() => {
-            ReactDOM.hydrate(<OuterApp {...props} />, rootEl, () => {
+            hydrateRoot(rootEl, <OuterApp {...props} />, () => {
```

Comment:
Just wanted to let you know that the API for `hydrateRoot` changed with React 18 and no callback is triggered. So you have to do this manually. [Source]( https://github.com/reactwg/react-18/discussions/5)

You can check the closed [React 18 poc PR](https://github.com/SalesforceCommerceCloud/pwa-kit/pull/925/files#diff-f4ee7217dbf638680d6780866b7e72f56790717a5c6cc3ed1c982794a180ae38) for a possible solution.